.github/workflows/server-tests.yml

```diff
@@ -1,6 +1,13 @@
 name: Server Tests
 
 on:
+    workflow_call:
+      inputs:
+        base_ref:
+          type : string
+      secrets:
+        codecov_token:
+          required: true
```

Comment:
Let's make codecov optional.
If this is available, only then we upload coverage data.